cli/internal/run/run.go

```diff
@@ -268,14 +272,19 @@ func (r *run) run(ctx gocontext.Context, targets []string) error {
 	}
 	packageManager := pkgDepGraph.PackageManager
 
-	engine, err := buildTaskGraphEngine(g, rs)
+	engine, err := buildTaskGraphEngine(
+		g,
+		rs,
+		r.opts.runOpts.singlePackage,
+	)
 
 	if err != nil {
 		return errors.Wrap(err, "error preparing engine")
 	}
 	tracker := taskhash.NewTracker(
 		g.RootNode,
 		g.GlobalHash,
+		// TODO(mehulkar): remove g,Pipeline, because we need to get task definitions from CompleteGaph instead
```

Comment:
> need

What is the need? Are there any implications for not doing this now? Or is it just a code cleanup thing?

Reply:
Th is passed in so the taskhash can include the task definition. With composable, we don't want to use the root definition, we'll want to use the _resolved_ definition. 

Since this work included reducing where / how we use `g.Pipeline` (i.e. the root pipeline), I was leaving a paper trail for myself for other usages of the root Pipeline that will disappear. [The compiler doesn't protect me from unused arguments](https://play.golang.com/p/lhzeLpiKsDj), so I wanted to be sure that I stopped passing it in when the `taskhash` internals are updated.